lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -2481,11 +2489,13 @@ class PrimalGenCloner final : public SILClonerWithScopes<PrimalGenCloner> {
     // or address operations to perform checkpointing.
 
     // Checkpoint nested primal values as a tuple.
-    auto nestedPrimValDeclTy =
-        joinElementTypesFromValues(primVals, getASTContext());
-    getPrimalInfo().addNestedStaticPrimalValueDecl(ai, nestedPrimValDeclTy);
-    auto primValAggr = joinElements(primVals, builder, primalCall->getLoc());
-    staticPrimalValues.push_back(primValAggr);
+    if (!primVals.empty()) {
+      auto nestedPrimValDeclTy =
+          joinElementTypesFromValues(primVals, getASTContext());
+      getPrimalInfo().addNestedStaticPrimalValueDecl(ai, nestedPrimValDeclTy);
+      auto primValAggr = joinElements(primVals, builder, primalCall->getLoc());
+      staticPrimalValues.push_back(primValAggr);
```

Comment:
should we release the elements in `primVals` after creating the AD checkpoint struct?

Reply:
No. Elements of the struct are to be kept live. Similarly, the following code results in no `release_value`.
```swift
import TensorFlow

struct T {
  var x, y: Tensor<Float>
}

func foo() -> Tensor<Float> {
  return Tensor(1)
}

func bar() -> T {
  return T(x: foo(), y: Tensor(1))
}
```

```swift
// bar()
sil hidden @$s3foo3barAA1TVyF : $@convention(thin) () -> @owned T {
  ...
  %14 = apply %13(%2, %11, %0) : $@convention(method) (@owned Tensor<Float>, @owned Tensor<Float>, @thin T.Type) -> @owned T // user: %15
  return %14 : $T                                 // id: %15
}
```